lib/render/RayCaster.cpp

```diff
@@ -511,6 +511,11 @@ int RayCaster::_initializeGL()
     GLint viewport[4];
     glGetIntegerv( GL_VIEWPORT, viewport );
     std::memcpy( _currentViewport, viewport, 4 * sizeof(GLint) );
+    // newViewport contains zero width and height sometimes. 
+    //   Need to filter out those instances.
+    for( int i = 2; i < 4; i++ )
```

Comment:
I think a better fix would be to NOT perform any viewport dependent initialization here. I.e. inside of _initializeFramebufferTextures don't do:

    glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA32F, _currentViewport[2], _currentViewport[3],
                 0, GL_RGBA, GL_FLOAT, nullptr);

glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA32F, _currentViewport[2], _currentViewport[3],
                 0, GL_RGBA, GL_FLOAT, nullptr);

Because you are using phony values and then correct them inside of _updateViewportWhenNecessary()

I.e. just call glTexImage2D inside of _updateViewportWhenNecessary

Reply:
Does your suggestion mean that `glTexImage2d()` needs to be called every `_paintGL()` is called? 
I definitely try to avoid that from happening, since viewport change is a pretty rare situation, and viewport being zero is more rare...